builder/builder.go

```diff
@@ -245,6 +247,13 @@ func (b *Builder) convertAndBuild(ctx context.Context, target domain.Target, opt
 				}
 
 				if !isMultiPlatform[saveImage.DockerTag] {
+					if singPlatImgNames[saveImage.DockerTag] {
```

Comment:
nit: since we're using `singPlatImgNames` as a set, perhaps this would make it more obvious:

```suggestion
					if _, found := singPlatImgNames[saveImage.DockerTag]; found {
```

it would also allow us to change to `singPlatImgNames := map[string]struct{}` down the road if we wanted to: https://itnext.io/set-in-go-map-bool-and-map-struct-performance-comparison-5315b4b107b

(but given the size of the set in this code, it's pre-mature optimization)